bigflow/deploy.py

```diff
@@ -35,18 +38,76 @@ def tag_image(image_id: str, repository: str, tag: str) -> str:
 
 
 def deploy_docker_image(
-        image_tar_path: str,
-        docker_repository: str,
-        auth_method: AuthorizationType = AuthorizationType.LOCAL_ACCOUNT,
-        vault_endpoint: T.Optional[str] = None,
-        vault_secret: T.Optional[str] = None,
+    image_tar_path: str,
+    docker_repository: str,
+    auth_method: AuthorizationType = AuthorizationType.LOCAL_ACCOUNT,
+    vault_endpoint: T.Optional[str] = None,
+    vault_secret: T.Optional[str] = None,
 ) -> str:
+    if image_tar_path.endswith(".toml"):
```

Comment:
If i understand correctly sometimes it is in fact path to tar_image but in other cases it is path not to image but to project_spec?

Reply:
No, it might be a path to 'image**.toml' file, which is a placeholder for an information about build.
This mostly exists to mimic old behaviour of bigflow "as much as possible".
For example now it is possible to build image, modify some local files (which result in changed version) and then deploy that image with "previous version". That is because now version is parsed from a file name, not from current project version. That change tries to keep existing behaviour.